mlflow/tracking/context/databricks_job_context.py

```diff
@@ -46,4 +53,28 @@ def tags(self):
             tags[MLFLOW_DATABRICKS_WORKSPACE_URL] = workspace_url_fallback
         if workspace_id is not None:
             tags[MLFLOW_DATABRICKS_WORKSPACE_ID] = workspace_id
+
+        git_repo_url = databricks_utils.get_git_repo_url()
+        git_repo_provider = databricks_utils.get_git_repo_provider()
+        git_repo_commit = databricks_utils.get_git_repo_commit()
+        git_repo_relative_path = databricks_utils.get_git_repo_relative_path()
+        git_repo_reference = databricks_utils.get_git_repo_reference()
+        git_repo_reference_type = databricks_utils.get_git_repo_reference_type()
+        git_repo_status = databricks_utils.get_git_repo_status()
+
+        if git_repo_url is not None:
+            tags[MLFLOW_DATABRICKS_GIT_URL] = git_repo_url
+        if git_repo_provider is not None:
+            tags[MLFLOW_DATABRICKS_GIT_PROVIDER] = git_repo_provider
+        if git_repo_commit is not None:
+            tags[MLFLOW_DATABRICKS_GIT_COMMIT] = git_repo_commit
+        if git_repo_relative_path is not None:
+            tags[MLFLOW_DATABRICKS_GIT_RELATIVE_PATH] = git_repo_relative_path
+        if git_repo_reference is not None:
+            tags[MLFLOW_DATABRICKS_GIT_REFERENCE] = git_repo_reference
+        if git_repo_reference_type is not None:
+            tags[MLFLOW_DATABRICKS_GIT_REFERENCE_TYPE] = git_repo_reference_type
+        if git_repo_status is not None:
+            tags[MLFLOW_DATABRICKS_GIT_STATUS] = git_repo_status
```

Comment:
Instead of replicating this logic across the job context and notebook context, I think it would make more sense to introduce a new `DatabricksRepoContext` that provides these tags if we're in a repo.